components/XOGame.js

```diff
@@ -0,0 +1,140 @@
+import styled from "styled-components";
+import {useState} from "react";
+
+const players = {
+
+  HUMAN: {
+  SYM: "X",
+  NAME: "You",
+  },
+};
+
+export default function XOGame() {
+  // const [board, setBoard] = useState(Array(9).fill(""));
+  const [board, setBoard] = useState([
+    ["", "", ""],
+    ["", "", ""],
+    ["", "", ""],
+  ]);
+
+  function playFn(arrayIndex, index) {
+    
+    
+    board[arrayIndex][index] = players?.HUMAN?.SYM;
+    setBoard((board) => [...board]);
+    
+    
+  }
+
+  function playAgainFn() {
+    setBoard([
+      ["", "", ""],
+      ["", "", ""],
+      ["", "", ""],
+    ]);
+   
+  }
+
+  
+  return (
+      <>
+        <Matchfield>
+          <Col />
+            <Cell onClick={() => playFn(0, 0)} >
+              {board[0][0]}
+            </Cell>
+            <Cell onClick={() => playFn(0, 1)} >
+              {board[0][1]}
+            </Cell>
+            <Cell onClick={() => playFn(0, 2)} >
+              {board[0][2]}
+            </Cell>
+          
+          <Col />
+            <Cell onClick={() => playFn(1, 0)} >
+              {board[1][0]}
+            </Cell>
+            <Cell onClick={() => playFn(1, 1)} >
+              {board[1][1]}
+            </Cell>
+            <Cell onClick={() => playFn(1, 2)} >
+              {board[1][2]}
+            </Cell>
+          
+          <Col />
+            <Cell onClick={() => playFn(2, 0)} >
+              {board[2][0]}
+            </Cell>
+            <Cell onClick={() => playFn(2, 1)} >
+              {board[2][1]}
+            </Cell>
+            <Cell onClick={() => playFn(2, 2)} >
+              {board[2][2]}
+            </Cell>
+          
+        </Matchfield>
+        
+        <Button onClick={playAgainFn}>
+            Restart
+        </Button>
+          
+        
+      </>
+      
+    );
+}
+
+const Matchfield = styled.section`
```

Comment:
It would be better to put the styling in another file than the main code

Reply:
I wanted to outsource the styling later if everything works, but thanks for the comment.